components/Pagination.tsx

```diff
@@ -33,75 +45,56 @@ export const Pagination = ({
     const buttonClassesActive = ['border-purple-200', 'text-purple-200'];
     const arrowButtons = ['w-full', 'h-full', 'flex', 'items-center'];
     const leftArrowClasses = ['transform', 'rotate-180'];
-    const url = `${baseUrl}/${search}`;
     // Pagination
     const totalPages = Math.ceil(totalCount / postPerPage);
-    const pagStart = PaginationStart(page, totalPages, paginationLimit);
-    const pagEnd = totalPages > paginationLimit ? paginationLimit : totalPages;
-    const buttonCount = new Array(pagEnd).fill(0);
+    const { buttonCount, pagStart } = PaginationLenght({
+        totalPages,
+        page,
+        paginationLimit
+    });
```

Comment:
This looks like the perfect candidate for you to learn how to create your own react hooks, I mean, you don't have to but I know you want it: https://reactjs.org/docs/hooks-custom.html 😂